pages/volunteer_role.tsx

```diff
@@ -0,0 +1,244 @@
+import { NavigateNextSharp } from '@mui/icons-material'
+import {
+  Box,
+  Breadcrumbs,
+  Button,
+  CircularProgress,
+  Link,
+  Stack,
+  Typography,
+  useTheme,
+} from '@mui/material'
+import SectionContainer from 'components/layout/SectionContainer'
+import { withBasicLayout } from 'components/layouts'
+import Masthead from 'components/Masthead'
+import { useEffect, useState } from 'react'
+import { DASVolunteerRole } from 'types'
+
+import { dasVolunteerRoleService } from './api/VolunteerRoleService'
+
+const VolunteerRolePage = () => {
+  const [role, setRole] = useState<DASVolunteerRole>()
+  const [loading, setLoading] = useState(true)
+
+  useEffect(() => {
+    const params = new URLSearchParams(window.location.search)
+    dasVolunteerRoleService
+      .getRoleDetailsByName(params.get('role'))
+      .then((role) => setRole(role))
+      .catch((error) => console.error(error))
```

Comment:
This is where the site should redirect to the 404 page if the role cannot be found.